qtpy/__main__.py

```diff
@@ -0,0 +1,3 @@
+import qtpy.cli
+
+qtpy.cli.cli()
```

Comment:
```suggestion

def main():
    qtpy.cli.cli()


if __name__ == "__main__":
    main()
```

I usually like to do things this way, as recommended in the setuptools docs, so I can point my setuptools entrypoint here so it doesn't break in case the cli file is moved, the function is renamed, or something else similar is changed.

Reply:
I don't see the major upside as compared with encouraging the hazardous behavior of making files both directly runnable and importable.

Reply:
I'm afraid I don't follow the reasoning here. Could you explain why this behavior is more hazardous in this context of a trivial module following a well-known common pattern, relative to the practical risk of having separate `console_scripts` and runpy entrypoints that may behave differently or get out of sync? The `if __name__ == "__main__"` idiom is very common (the [question](https://stackoverflow.com/questions/419163/what-does-if-name-main-do) explaining how and why to use it is the second most popular tagged `python` on all of SO), especially in the scientific Python ecosystem, and is recommended in the [official Python docs](https://docs.python.org/3/library/__main__.html) and countless other sources.

Further, this particular pattern is recommended a [substantial](https://chriswarrick.com/blog/2014/09/15/python-apps-the-right-way-entry_points-and-scripts/) [number](https://stackoverflow.com/questions/27784271/how-can-i-use-setuptools-to-generate-a-console-scripts-entry-point-which-calls) of [places](https://stackoverflow.com/questions/44977227/how-to-configure-main-py-init-py-and-setup-py-for-a-basic-package), none of which describe any such issues with it.

Of course, there is the general principle of separation of concerns, and thus that entry points and package functionality should be separate (which this still fully ensures, as this module contains no non-trivial functionality beyond that necessary to serve as a single-source entrypoint for both `console_scripts` and `runpy`). 

Beyond that, the only directly relevant post I found explicitly advocating against any use of `if __name__ == __main__` was [this personal blog](https://moshez.wordpress.com/2016/06/07/__name__-__main__-considered-harmful/), the one comment to which was a rebuttal by a Python core dev, and the (rather esoteric) potential harms mentioned don't apply to a module containing no non-trivial functionality beyond re-directing to the entry point. Some of these are at least theoretically possible if `qtpy,cli` were made runnable, but that's not what is being proposed here, and following your suggestions above, I rejected that. Therefore, I cannot see what the practical harm is, relative to the practical benefit (and avoidance of much more likely harm) of single-sourcing the entry piont.